Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/CorelliPowderCalibrationCreate.py

```diff
@@ -0,0 +1,576 @@
+
+
+def unique_workspace_name(n: int = 5, prefix: Optional[str] = '',
+                          suffix: Optional[str] = ''):
+    r"""
+    Create a random sequence of `n` lowercase characters that is guaranteed
+    not to collide with the name of any existing Mantid workspace registered
+    in the analysis data service.
+
+    @param n : size of the sequence
+    @param prefix : string to prefix the randon sequence
+    @param suffix : string to suffix the randon sequence
+    """
+
+    n_seq = ''.join(random.choice(string.ascii_lowercase) for _ in range(n))
+    ws_name = '{}{}{}'.format(str(prefix), n_seq, str(suffix))
+    while ws_name in AnalysisDataService.getObjectNames():
+        characters = [random.choice(string.ascii_lowercase) for _ in range(n)]
+        n_seq = ''.join(characters)
+        ws_name = '{}{}{}'.format(str(prefix), n_seq, str(suffix))
+    return ws_name
+
+
+def temp_workspace_generator(temp_workspaces):
+    r"""
+    Create a funtion that serve a unique workspace name and position it to be later deleted
+
+    @param temp_workspaces :: list of workspaces to be deleted after algorithm has finished
+
+    @return function that serves the workspace names
+    """
+    def inner_function():
+        name = unique_workspace_name()
+        temp_workspaces.append(name)
+        return name
+    return inner_function
+
+
+def insert_bank_numbers(input_workspace: Union[str, Workspace2D],
+                        grouping_workspace: Union[str, WorkspaceGroup]):
+    r"""
+    Label each spectra according to each bank number
+
+    @param input_workspace : workpace with spectra to be labelled
+    @param grouping_workspace : group workspace has the group ID (bank number) for each pixel
+    """
+    input_handle, grouping_handle = mtd[str(input_workspace)], mtd[str(grouping_workspace)]
+    group_ids = sorted(list(set(grouping_handle.extractY().flatten())))
+    assert input_handle.getNumberHistograms() == len(group_ids)
+    axis = TextAxis.create(len(group_ids))
+    [axis.setLabel(index, f'bank{group_id}') for index, group_id in enumerate(group_ids)]
+    input_handle.replaceAxis(1, axis)
+
+
+class CorelliPowderCalibrationCreate(DataProcessorAlgorithm):
+
+    peak_shapes = ['Gaussian']
+    bank_count = 92
+    _banks = [f'bank{i}/sixteenpack' for i in range(1, bank_count)]
+    r"""The calibration attempts to adjust the location and orientation of various instrument components, namely
+    the source and detector banks. The items describing a full adjustment are as follows:
+    - Component: name of the instrument component (e.g. 'moderator', 'bank42/sixteenpack')
+    - Xposition, YPosition, ZPosition: location of the instrument component in the lab frame (units in meters)
+    - XdirectionCosine, YdirectionCosine, ZdirectionCosine, RotationAngle: direction cosines and rotation angle
+      (in degress) defining a rotation in the lab frame that orients the instrument component
+    """
```

Comment:
If the docstring of a class if provided after class attributes, can Python interpreter still treat the docstring as the class's docstring?

Reply:
I wanted to document just `adjustment_items` but `r""".."""` is not right. I should be using `#:`. Changing it now